demystifying_regular_expressions/demystifying_regular_expressions.md

```diff
@@ -0,0 +1,305 @@
+<!--
+
+author:   Joy Payton
+email:    paytonk@chop.edu
+version:  1.0.0
+current_version_description: Initial version
+module_type: standard
+docs_version: 1.0.0
+language: en
+narrator: UK English Female
+mode: Textbook
+
+title: Demystifying Regular Expressions
+
+comment:  Learn about pattern matching using regular expressions, or regex.
+
+long_description: Regular expressions, or regex, are a way to specify patterns (such as the pattern that defines a valid email address, medical record number, or credit card number) within text.  Learn about how regular expressions can move your research forward in this non-coding module.
+
+estimated_time_in_minutes: 30
+
+@pre_reqs
+This module does not require any particular knowledge.  Anyone who has used a search function to find or find and replace text in a document will be able to engage with this content.
+@end
+
+@learning_objectives  
+After completion of this module, learners will be able to:
+
+- Explain what a regular expression is 
+- Give an example of how regular expressions can be useful
+- Use an online regular expressions checker that helps build and test regular expressions.
+
+@end
+
+@version_history 
+No previous versions.
+@end
+
+import: https://raw.githubusercontent.com/arcus/education_modules/main/_module_templates/macros.md
+-->
+
+# Demystifying Regular Expressions
+
+@overview
+
+## Regular Expressions, Defined
+
+The word *regular* (like *regulate*) comes from the Latin root for "rule": *regula*.  A **regular expression** is a specific way to **express** a **rule** for a pattern.  
+
+For example, one pattern rule might be for the URL of a root web page, like https://www.nytimes.com or http://apache.org.  
+
+Other rules might describe the pattern of a valid credit card number, or a national identity number, or a postal address.  
+
+As humans, we are very good at identifying patterns and realizing instantly that a certain text "looks like an email address" or "looks like a phone number".  Quantifying patterns and telling the computer what a pattern "looks like" is what regular expressions are all about.
+
+<h3>Patterns in Biomedical Research</h3>
+
+In biomedical research, patterns emerge in a number of ways:
+
+* File names might follow a pattern that include a subject identifier and date stamp, which can be useful in filtering out only file names from a certain time period or for a certain subject.
+* Clinical notes in an electronic health record might contain text like " R/O ", " RO:", " r/o " which mean "rule out" or "ruled out", which are important to find in order to discover differential diagnosis clues.
```

Comment:
This sentence is all of my grammar/comma pet peeves, feel free to just say "stop pestering me about this it isn't important" but here is a suggestion to "fix" the comma problems... or at least give me less anxiety about commas going inside or outside of quotes.

```suggestion
* Clinical notes in an electronic health record might represent the phrases "rule out" or "ruled out" as `R/O`, `RO:`,  or `r/o`. It is important to find all of these in order to discover differential diagnosis clues.
```